dev/src/bulk-writer.ts

```diff
@@ -1028,6 +1039,19 @@ export class BulkWriter {
     }
   }
 
+  /**
+   * Silences the promise if an error handler has been passed in. This prevents
+   * UnhandledPromiseRejections from being thrown if a floating BulkWriter
+   * operation promise fails.
+   * @private
+   * @internal
+   */
+  private _silencePromiseIfErrorHandlerSet(promise: Promise<unknown>): void {
+    if (this._errorHandlerSet) {
```

Comment:
Can we do this slightly later and make this decision when the operation returns? This would allow the following:

```
bulkWriter.set();
bulkWriter.set();
bulkWriter.onError(...);
```

Note that this error handler will always be attached before any result is returned.

Reply:
What's the use case for changing the error handler after an operation has been enqueued but before it resolves? To me, it seems like an anti-pattern, or an accidental bug.

Reply:
It's guaranteed that this code executes before the request is processed, so I don't think we should forbid this. It is not possible for any network task to resolve before all synchronous code finished executing. I do think it is cleaner to attach the error handler first, but we do not enforce this, so I think we should allow this even if we believe this to be an anti-pattern.